libs/preallocate/src/main/java/org/elasticsearch/preallocate/PreallocateModuleExportsService.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.preallocate;
+
+import org.elasticsearch.jdk.ModuleQualifiedExportsService;
+
+public class PreallocateModuleExportsService extends ModuleQualifiedExportsService {
```

Comment:
While a little unfortunate that we need to do this in all target modules, it's simple and straightforward.

Reply:
And very much auditable. Or even generate-able if we wanted, in the future.

Reply:
One last comment on this ( and again for the future ). What you have is good, but as an alternative we could reduce this to a single method that returns a `MethodHandles.Lookup`, that holds the PRIVATE access mode of the declaring module. The service can then use that lookup to perform invocations of caller-sensitive methods as-if it were the caller.  see https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/invoke/MethodHandles.Lookup.html#callsens

Reply:
I like that idea, but I will leave it for the future.